plugin-server/src/main/ingestion-queues/kafka-queue.ts

```diff
@@ -220,7 +112,18 @@ export class KafkaQueue implements Queue {
         return Promise.resolve()
     }
 
-    resume(targetTopic: string = this.pluginsServer.KAFKA_CONSUMPTION_TOPIC!, partition?: number): void {
+    async sleep(sleepMs: number, partition: number, topic = this.bufferTopic): Promise<void> {
+        this.sleepTimeout = setTimeout(() => {
```

Comment:
I see what this is doing - we make sure we don't sleep more than sleepMs even if pausing takes N units in time.

However isn't this just being too clever?

If not, a better way is this to avoid messing with timeouts and promises in the same code:


```
await Promise.all([
  this.pause,
  delay(sleepMs).then(() => this.resume())
])
```

Reply:
I believe we use this pattern everywhere - also we don't want to await that delay